### PR TITLE
FAT-732: Implement e-2-e automation of test case C641

### PR DESCRIPTION
**Pull request by Stanislau-Snapok**
## Purpose
Implement e-2-e automation of test case C641.

## Approach
I used `cy.wait` twice:
- for 2000ms when updating schedule. Without `wait` sometime it shows seccessfull toast that schedule was updated but data for it was not realy changed.
- for 5000ms before click on user loans in checkout app. There are a lot of requests within user loading, and if leave this page before all of them got responce, test will fail by reason of aborting of requests.

## Refs
https://issues.folio.org/browse/FAT-732

## Screenshots
image

### CODE REVIEW

File: cypress/support/fragments/checkout/checkout.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export default {
+  fillUserBarcode(barcode) {
+    cy.do([
+      TextField({ id: 'input-patron-identifier' }).fillIn(barcode),
+      Button({ id: 'clickable-find-patron' }).click(),
+    ]);
+  },
+  checkUserOpenLoans({ barcode, id }) {
+    this.fillUserBarcode(barcode);
+    cy.wait(5000);
```
Comment (dmitriyfolio) [inline cypress/support/fragments/checkout/checkout.js:15]: "for 5000ms before click on user loans in checkout app. There are a lot of requests within user loading, and if leave this page before all of them got responce, test will fail by reason of aborting of requests." - if it's related with this  description - you need to wait expected responses instead

File: cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+          },
+          patronGroup: Cypress.env(CY_ENV.USER_GROUPS)[0].id,
+          departments: [],
+        })
+          .then((user) => {
+            userId = user.id;
+          });
+        cy.createInstance({
+          instance: {
+            instanceTypeId: Cypress.env(CY_ENV.INSTANCE_TYPES)[0].id,
```
Comment (dmitriyfolio) [inline cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js:63]: try to work through test consts instead of Cypress.env

File: cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+          rulesAsText: rulesDefaultString,
+        });
+        cy.deleteLoanPolicy(Cypress.env(CY_ENV.LOAN_POLICY).id)
+          .then(() => {
+            cy.deleteFixedDueDateSchedule(Cypress.env(CY_ENV.FIXED_DUE_DATE_SCHEDULE).id);
+          });
+      });
+  });
+
+  it('C641: renewing item using a fixed due date loan profile where the fixed due date schedule date range does not cover the test date',
```
Comment (dmitriyfolio) [inline cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js:158]: try to move these points into api steps in related fragements:
"Create a fixed due date schedule that covers the test date
Create a loan policy using that schedule
Create a circulation rule using the loan policy
Check out an item entering a patron and item that meet the circulation rule "

File: cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      cy.visit(SettingsMenu.circulationFixedDueDateSchedulesPath);
+      fixedDueDateSchedules.editSchedule(Cypress.env(CY_ENV.FIXED_DUE_DATE_SCHEDULE).name, {
+        description: Cypress.env(CY_ENV.FIXED_DUE_DATE_SCHEDULE).description,
+        schedules:[{
+          from: fromDate.format('YYYY/MM/DD'),
+          to: newToDate.format('YYYY/MM/DD'),
+          due: dueDate.format('YYYY/MM/DD'),
+        }],
+      });
+      cy.contains(`${newToDate.format('M/D/YYYY')}`).should('be.visible');
```
Comment (dmitriyfolio) [inline cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js:170]: wrap to method with meaning name into appropriate fragment

File: cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+      cy.visit(topMenu.checkOutPath);
+      checkout.checkUserOpenLoans(Cypress.env(CY_ENV.USER));
+      cy.contains(`${Cypress.env(CY_ENV.LOAN_POLICY).name}`).should('be.visible');
```
Comment (dmitriyfolio) [inline cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js:173]: wrap to method with meaning name into appropriate fragment

File: cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+      loans.renewLoan();
+      cy.contains('renewal date falls outside of date ranges in fixed loan policy').should('be.visible');
```
Comment (dmitriyfolio) [inline cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js:175]: wrap to method with meaning name into appropriate fragment

File: cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+  });
+
+  it('C641: renewing item using a fixed due date loan profile where the fixed due date schedule date range does not cover the test date',
+    { tags: [TestTypes.smoke] },
+    () => {
+      cy.visit(SettingsMenu.circulationFixedDueDateSchedulesPath);
+      fixedDueDateSchedules.editSchedule(Cypress.env(CY_ENV.FIXED_DUE_DATE_SCHEDULE).name, {
+        description: Cypress.env(CY_ENV.FIXED_DUE_DATE_SCHEDULE).description,
+        schedules:[{
+          from: fromDate.format('YYYY/MM/DD'),
```
Comment (dmitriyfolio) [inline cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js:165]: keep special formats into related fragments

File: cypress.json [hunk 8]
```diff
@@ -1,17 +1,17 @@
...
-  "baseUrl": "https://folio-snapshot.dev.folio.org",
+  "baseUrl": "https://folio-snapshot-2.dev.folio.org",
   "viewportWidth": 1920,
   "viewportHeight": 1080,
   "video": false,
   "defaultCommandTimeout": 151000,
   "pageLoadTimeout": 120000,
   "env": {
-    "OKAPI_HOST": "https://folio-snapshot-okapi.dev.folio.org",
+    "OKAPI_HOST": "https://folio-snapshot-2-okapi.dev.folio.org",
```
Comment (dmitriyfolio) [inline cypress.json:9]: revert this change

File: cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js (hunk 7 above)
Comment (Stanislau-Snapok) [reply cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js]: Done

File: cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js (hunk 6 above)
Comment (Stanislau-Snapok) [reply cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js]: Done

File: cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js (hunk 3 above)
Comment (Stanislau-Snapok) [reply cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js:176]: I moved some common thing inside methods, but looks like the most of thing are very specific to do fragments from them.

File: cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js (hunk 2 above)
Comment (Stanislau-Snapok) [reply cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js:75]: Refactored, but not everywhere. Because a lot of values used once but in different scope than where they can be assigned. So don't want to create a lot of `let` variables for these values.

File: cypress/support/fragments/checkout/checkout.js (hunk 1 above)
Comment (Stanislau-Snapok) [reply cypress/support/fragments/checkout/checkout.js]: I tried what we discussed but it didn't work. I left `cy.wait` here but decreased timer to 3500.

File: cypress.json (hunk 8 above)
Comment (Stanislau-Snapok) [reply cypress.json]: Done
